docs/UserGuide.md

```diff
@@ -1,186 +1,239 @@
+> INDEX_NUMBER of task is shown as according to the current tab.
 
-Deletes the specified person from the address book. Irreversible.<br>
-Format: `delete INDEX`
+Examples:
+* `complete 2`
+* `complete 5`
 
-> Deletes the person at the specified `INDEX`. <br>
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+### 2.8. Track overall progress: `progress`
 
-Examples:
+Shows all the tasks completed over time and also specifying the tasks overdued.<br>
+> A general overview of performance/efficiency based on the usage of the task manager will be shown.<br>
 
-* `list`<br>
-  `delete 2`<br>
-  Deletes the 2nd person in the address book.
-* `find Betsy`<br>
-  `delete 1`<br>
-  Deletes the 1st person in the results of the `find` command.
+Format: `progress NUMBER_OF_DAYS`
+> NUMBER_OF_DAYS is a interger which indicates the your progress for these past days (excluding today).<br>
+> It will provide an analysis on the number of task that you have completed/overdue.
 
-### 2.7. Select a person : `select`
+Examples:
+* `progress 7`
 
-Selects the person identified by the index number used in the last person listing.<br>
-Format: `select INDEX`
+### 2.9. Search for tasks: `search`
 
-> Selects the person and loads the Google search page the person at the specified `INDEX`.<br>
-> The index refers to the index number shown in the most recent listing.<br>
-> The index **must be a positive integer** 1, 2, 3, ...
+Searches for tasks based on keywords by user's input.<br>
 
-Examples:
+Format: `search KEYWORD`
+> A valid input needs to be keyed in. No special characters such as ASCII whitespace is allowed.<br>
+> A popout will appear to show you the list of tasks that contains the stated `KEYWORD`
 
-* `list`<br>
-  `select 2`<br>
-  Selects the 2nd person in the address book.
-* `find Betsy` <br>
-  `select 1`<br>
-  Selects the 1st person in the results of the `find` command.
+Examples:
+* `search potato`
 
-### 2.8. Clearing all entries : `clear`
+### 2.10. Undo previous action : `undo`
 
-Clears all entries from the address book.<br>
-Format: `clear`
+Undos previous action that user made.<br>
+Format : `undo`
+> Any previous valid input will be reversed.
+> Only 1 input will be reversed at a time.
 
-### 2.9. Exiting the program : `exit`
+### 2.11. Redo previous action : `redo`
 
-Exits the program.<br>
-Format: `exit`
+Redos previous `undo` that user made.<br>
+Format : `redo`
+> Any previous `undo` will be reversed.
 
-### 2.10. Saving the data
+### 2.12. Saving the tasks
 
-Address book data are saved in the hard disk automatically after any command that changes the data.<br>
-There is no need to save manually.
+Upon creation of tasks, the tasks will be automatically saved in the folder where the program is held in.
+> Do not erase the saved data as it will result in a complete loss of data that cannot be recovered by the application itself.
```

Comment:
If the tasks are saved after every mutable command, then it does not really matter that the data erased, as the next command will save a new file.  Or are you mentioning about erasing the data before executing the program?   

You also missed out on the feature to save in a different file/directory.